conditions.go

```diff
@@ -88,14 +89,46 @@ func (a Address) Equals(b Address) bool {
 // MarshalJSON provides a hex representation for JSON,
 // to override the standard base64 []byte encoding
 func (a Address) MarshalJSON() ([]byte, error) {
-	return marshalHex(a)
+	s := strings.ToUpper(hex.EncodeToString(a))
+	return json.Marshal(s)
 }
 
-// UnmarshalJSON parses JSON in hex representation,
-// to override the standard base64 []byte encoding
-func (a *Address) UnmarshalJSON(src []byte) error {
-	dst := (*[]byte)(a)
-	return unmarshalHex(src, dst)
+func (a *Address) UnmarshalJSON(raw []byte) error {
+	var enc string
+	if err := json.Unmarshal(raw, &enc); err != nil {
+		return errors.Wrap(err, "cannot decode json")
+	}
+
+	// If the encoded string starts with a prefix, cut it off and use
+	// specified decoding method instead of default one.
+	chunks := strings.SplitN(enc, ":", 2)
+	format := chunks[0]
+	if len(chunks) == 1 {
+		format = "hex"
```

Comment:
Don't you have to set `enc = chunks[0]` here?

Otherwise, only `hex:F001234...` will be valid, and not `F001234...`

Reply:
Nevermind, you read into enc above...

I got used to not reusing variables in the iov-core project (linter prohibits it mostly), but code is fine

Reply:
This is on purpose. If there is only one chunk, there is no `:` character and this is the old encoding format - hex without the encoding prefix.
Otherwise, there is both prefix and value.